### PR TITLE
MCNP6 support

### PR DESCRIPTION
**Pull request by ljacobson64**
This PR adds DAGMC integration for MCNP6. The bulk of the changes is the addition of the two patch files for versions 6.1 and 6.1.1. Significant changes were also made to mcnp_funcs.cpp to accommodate more types of metadata (MCNP6 particle types, FCL, etc.) in the gcad files.

### CODE REVIEW

File: common/CMakeLists.txt [hunk 1]
```diff
@@ -16,6 +16,4 @@ add_library(dagmciface ${LIB_TYPE} ${DI_SRC_FILES})
 set_target_properties(dagmciface PROPERTIES PUBLIC_HEADER "${DI_PUB_HEADERS}")
 install(TARGETS dagmciface LIBRARY       DESTINATION ${INSTALL_LIB_DIR}
                            ARCHIVE       DESTINATION ${INSTALL_LIB_DIR}
-                           PUBLIC_HEADER DESTINATION ${INSTALL_INCLUDE_DIR}
-                           LIBRARY       PERMISSIONS ${PERMS}
-                           ARCHIVE       PERMISSIONS ${PERMS})
+                           PUBLIC_HEADER DESTINATION ${INSTALL_INCLUDE_DIR})
```
Comment (makeclean) [inline common/CMakeLists.txt]: these changes are unrelated to mcnp6, should they be in the pr?

File: fluka/CMakeLists.txt [hunk 2]
```diff
@@ -29,8 +29,7 @@ set_target_properties(fludag PROPERTIES PUBLIC_HEADER ${PUB_HEADERS})
 target_link_libraries(fludag ${LINK_LIBS})
 install(TARGETS fludag LIBRARY       DESTINATION ${INSTALL_LIB_DIR}
                        ARCHIVE       DESTINATION ${INSTALL_LIB_DIR}
-                       PUBLIC_HEADER DESTINATION ${INSTALL_INCLUDE_DIR}
-                       LIBRARY       PERMISSIONS ${PERMS})
+                       PUBLIC_HEADER DESTINATION ${INSTALL_INCLUDE_DIR})
```
Comment (makeclean) [inline fluka/CMakeLists.txt]: these changes are unrelated to mcnp6, should they be in the pr?

File: fluka/build/CMakeLists.txt [hunk 3]
```diff
@@ -18,5 +18,4 @@ include_directories(${MOAB_INCLUDE_DIRS})
 # Build FluDAG
 add_executable (mainfludag ${SRC_FILES})
 target_link_libraries(mainfludag ${LINK_LIBS})
-install(TARGETS mainfludag DESTINATION ${INSTALL_BIN_DIR}
-                           PERMISSIONS ${PERMS})
+install(TARGETS mainfludag DESTINATION ${INSTALL_BIN_DIR})
```
Comment (makeclean) [inline fluka/build/CMakeLists.txt]: these changes are unrelated to mcnp6, should they be in the pr?

File: geant4/CMakeLists.txt [hunk 4]
```diff
@@ -32,9 +32,7 @@ set_target_properties(dagsolid PROPERTIES PUBLIC_HEADER ${PUB_HEADERS})
 target_link_libraries(dagsolid ${LINK_LIBS})
 install(TARGETS dagsolid LIBRARY       DESTINATION ${INSTALL_LIB_DIR}
                          ARCHIVE       DESTINATION ${INSTALL_LIB_DIR}
-                         PUBLIC_HEADER DESTINATION ${INSTALL_INCLUDE_DIR}
-                         LIBRARY       PERMISSIONS ${PERMS}
-                         ARCHIVE       PERMISSIONS ${PERMS})
+                         PUBLIC_HEADER DESTINATION ${INSTALL_INCLUDE_DIR})
```
Comment (makeclean) [inline geant4/CMakeLists.txt]: these changes are unrelated to mcnp6, should they be in the pr?

File: gtest/CMakeLists.txt [hunk 5]
```diff
@@ -24,6 +24,4 @@ add_definitions(-DGTEST_HAS_PTHREAD=ON)
 # Build the static gtest library
 add_library(gtest STATIC ${SRC_FILES})
 install(TARGETS gtest LIBRARY       DESTINATION ${INSTALL_LIB_DIR}
-                      ARCHIVE       DESTINATION ${INSTALL_LIB_DIR}
-                      LIBRARY       PERMISSIONS ${PERMS}
-                      ARCHIVE       PERMISSIONS ${PERMS})
+                      ARCHIVE       DESTINATION ${INSTALL_LIB_DIR})
```
Comment (makeclean) [inline gtest/CMakeLists.txt]: these changes are unrelated to mcnp6, should they be in the pr?

File: mcnp/mcnp6/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+
+# Preprocessor directives
+add_definitions(-DKODE="mcnp")
+if (MPI_BUILD)
+  add_definitions(-DVERS="6.mpi")
+else ()
+  add_definitions(-DVERS="6")
+endif (MPI_BUILD)
+if (${MCNP_VERSION} STREQUAL 6.1)
+  add_definitions(-DTHREAD="MCNP6")
```
Comment (makeclean) [inline mcnp/mcnp6/CMakeLists.txt:121]: is the difference between the 6.1 branch of this if statement and the 6.1.1 branch to maintain parity with the LANL version, or should we make these more consistent?

File: uwuw/CMakeLists.txt [hunk 7]
```diff
@@ -7,25 +7,38 @@ set(DI_SRC_FILES ${DI_SRC_FILES} ${CMAKE_CURRENT_LIST_DIR}/uwuw.cpp PARENT_SCOPE
...
+set(SRC_FILES "uwuw_preproc.cpp")
+add_executable(uwuw_preproc ${SRC_FILES})
+list(APPEND LINK_LIBS uwuw_preprocessor)
+target_link_libraries(uwuw_preproc ${LINK_LIBS})
+install(TARGETS uwuw_preproc DESTINATION ${INSTALL_BIN_DIR})
 
 # Build UWUW tests
 add_subdirectory(tests)
-
-install(TARGETS uwuw_preproc DESTINATION bin)
```
Comment (makeclean) [inline uwuw/CMakeLists.txt:31]: this seems very unrelated to mcnp6

File: tools/make_watertight/CMakeLists.txt [hunk 8]
```diff
@@ -39,13 +39,10 @@ add_library(makeWatertight ${LIB_TYPE} Zip.cpp Arc.cpp Cleanup.cpp Gen.cpp Check
...
 
 # Install
-install(TARGETS check_watertight make_watertight DESTINATION ${INSTALL_BIN_DIR} PERMISSIONS ${PERMS})
-install(TARGETS makeWatertight LIBRARY 	     DESTINATION ${INSTALL_LIB_DIR}
-			       ARCHIVE       DESTINATION ${INSTALL_LIB_DIR}
-			       PUBLIC_HEADER DESTINATION ${INSTALL_INCLUDE_DIR}/make_watertight
-			       LIBRARY       PERMISSIONS ${PERMS}
-			       ARCHIVE 	     PERMISSIONS ${PERMS})
-
+install(TARGETS check_watertight make_watertight DESTINATION ${INSTALL_BIN_DIR} PERMISSIONS)
```
Comment (makeclean) [inline tools/make_watertight/CMakeLists.txt]: these changes are unrelated to mcnp6, should they be in the pr?

**Review by makeclean — CHANGES_REQUESTED** [review]
I think there is a little git work to do to remove the commits that arent part of the mcnp/mcnp6 work. Also the question regarding the if statements.

File: mcnp/mcnp6/CMakeLists.txt (hunk 6 above)
Comment (ljacobson64) [reply mcnp/mcnp6/CMakeLists.txt:121]: The differences are to maintain parity with the LANL versions. I copied those variables directly from `mcnp_env.F90`, which is a source file produced by the perl script `mcnp_env.pl`. The values of those variables in the CMakeLists are what appeared in `mcnp_env.F90` for versions 6.1 and 6.1.1.

Upon further investigation, the script `VC_info.gcf` contains `THREAD = MCNP6` and `REL_VERS = 1.0` for 6.1 and `THREAD = MCNP` and `REL_VERS = 6.1.1b` for 6.1.1.

**Comment by ljacobson64** [comment]
The extraneous non-MCNP6 commits should be gone now

**Comment by ljacobson64** [comment]
I'll preemptively answer the question about why I renamed the patch files: if you end them with `.patch`, you get syntax highlighting on github

**Comment by makeclean** [comment]
Thanks @ljacobson64, this is great :)

**Pull request merged**